### DISCUSSION TITLE
Template literal not supported in Typescript version < 5.0

### DISCUSSION THREAD

**Discussion by nicolas-c-dev** [floating-ui/floating-ui#2673] [Q&A] [answered]
Hi guys, i'm working on a project using typescript v4.0.3 and i'm having a problem with one type on the library that uses a template literal. The situation is that TS supports template literals since v5 and I need to override the type to be able to compile the app.
![image](https://github.com/floating-ui/floating-ui/assets/8958987/3fb4a429-9e0e-4d74-843f-7d8ae436657a)

To make this, I declared the new type in a .d.ts file:
![image](https://github.com/floating-ui/floating-ui/assets/8958987/726c40cf-8cf1-4df3-a860-45214ae6e84f)

But that generates another error in this type and I don't know how to continue. Any help will be welcomed! thanks!
![image](https://github.com/floating-ui/floating-ui/assets/8958987/76e42f76-6704-400e-8476-fac64a78e699)

**Comment by atomiks** [comment, marked as answer]
This library's minimum TS version is 4.1 - shouldn't be hard to increment your TypeScript version by 1 minor, in terms of breakages. 

Keep in mind, that version is over 3 years old now and has been the min TS version for this library since its release in Dec. 2021.

**Comment by nicolas-c-dev** [comment]
Got it, but i'm using NX cause this project is a monorepo so if I upgrade TS, i must upgrade eslint, and many other packages related to NX and its expensive. I think ther should be a work arround to override this only type.

**Reply by FezVrasta** [reply]
You can use patch-package with npm, or the built in package patcher with Yarn.

**Comment by nicolas-c-dev** [comment]
I just discovered that upgrading one minor version (typescript 4.1 implements template literals) and one major of @typescript-eslint/eslint-plugin solved the problem. Thanks for the help anyway.
